test/dotnet.Tests/CommandTests/ToolRestoreCommandWithMultipleNugetConfigTests.cs

```diff
@@ -0,0 +1,189 @@
+                _fileSystem,
+                toolPackageStoreMock,
+                new ProjectRestorerMock(
+                    _fileSystem,
+                    _reporter,
+                    new[]
+                    {
+                        new MockFeed
+                        {
+                            Type = MockFeedType.FeedFromLookUpNugetConfig,
+                            Uri = _nugetConfigUnderTestRoot,
+                            Packages = new List<MockFeedPackage>
+                            {
+                                new MockFeedPackage
+                                {
+                                    PackageId = _packageIdA.ToString(),
+                                    Version = _packageVersionA.ToNormalizedString(),
+                                    ToolCommandName = _toolCommandNameA.ToString()
+                                },
+                            }
+                        },
+
+                        new MockFeed
+                        {
+                            Type = MockFeedType.FeedFromLookUpNugetConfig,
+                            Uri = _nugetConfigUnderSubDir,
+                            Packages = new List<MockFeedPackage>
+                            {
+                                new MockFeedPackage
+                                {
+                                    PackageId = _packageIdB.ToString(),
+                                    Version = _packageVersionB.ToNormalizedString(),
+                                    ToolCommandName = _toolCommandNameB.ToString()
+                                },
+                            }
+                        }
+                    }));
+        }
+
+        [Fact]
+        public void WhenManifestPackageAreFromDifferentDirectoryItCanFindTheRightNugetConfigAndSaveToCache()
+        {
+            IToolManifestFinder manifestFileFinder =
+                new MockManifestFileFinder(new[]
+                {
+                    new ToolManifestPackage(_packageIdA, _packageVersionA,
+                        new[] {_toolCommandNameA},
+                        new DirectoryPath(Path.GetDirectoryName(_nugetConfigUnderTestRoot))),
+                    new ToolManifestPackage(_packageIdB, _packageVersionB,
+                        new[] {_toolCommandNameB},
+                        new DirectoryPath(Path.GetDirectoryName(_nugetConfigUnderSubDir)))
+                });
+
+            ToolRestoreCommand toolRestoreCommand = new ToolRestoreCommand(_appliedCommand,
+                _parseResult,
+                _toolPackageInstallerMock,
+                manifestFileFinder,
+                _localToolsResolverCache,
+                _fileSystem,
+                _nugetGlobalPackagesFolder,
+                _reporter
+            );
+
+            toolRestoreCommand.Execute().Should()
+                .Be(0, "if nuget prob from sub dir, it will find only the nuget.config under sub dir. " +
```

Comment:
Nit: typo. 
```suggestion
                .Be(0, "if nuget probed from sub dir, it will find only the nuget.config under sub dir. " +
```

Reply:
fixed, i took your suggestion and found and fixed other "prob"s